eventnotifier/slack/slack.go

```diff
@@ -96,105 +107,178 @@ func (n *Notifier) GetNotifierName() string {
+			logger.Warnw("slack notifier: couldn't send a message to the channel from the reporting", "slackChannel", dis.Spec.Reporting.SlackChannel, "error", err)
+		}
 	}
 
 	// We expect notification equal to or above success to be sent to users
 	if !types.NotificationSuccess.Allows(notifType) {
-		n.logger.Debugw("slack notifier: not sending info notification type to not flood user", "disruptionName", dis.Name, "eventType", event.Type, "message", bodyText)
+		logger.Debugw("slack notifier: not sending info notification type to not flood user", "message", slackMsg.BodyText)
 
 		return nil
 	}
 
-	emailAddr, err := mail.ParseAddress(userInfo.Username)
-	if err != nil {
-		return fmt.Errorf("slack notifier: invalid user info email in disruption %s: %w", dis.Name, err)
+	if err := n.sendMessageToUserChannel(dis, slackMsg, logger); err != nil {
+		logger.Warnw("slack notifier: couldn't send a message to the user", "error", err)
+
+		return fmt.Errorf("slack notifier: %w", err)
 	}
 
-	p1, err := n.client.GetUserByEmail(emailAddr.Address)
-	if err != nil {
-		n.logger.Warnw("slack notifier: user not found", "userAddress", emailAddr.Address, "error", err)
-		return nil
+	logger.Debugw("notifier: sending notifier event to slack", "message", slackMsg.BodyText)
+
+	return nil
+}
+
+func (n *Notifier) notifyForDisruptionCron(disruptionCron *v1beta1.DisruptionCron, event corev1.Event, notifType types.NotificationType) error {
+	logger := n.logger.With("disruptionCronName", disruptionCron.Name, "eventType", event.Type)
+
+	slackMsg := n.buildSlackMessage(disruptionCron, event, notifType, disruptionCron.Spec.Reporting, logger)
+
+	if n.config.MirrorSlackChannelID != "" {
+		if err := n.sendMessageToChannel(n.config.MirrorSlackChannelID, slackMsg); err != nil {
+			logger.Warnw("slack notifier: couldn't send a message to the mirror slack channel", "slackChannel", n.config.MirrorSlackChannelID, "error", err)
+		}
 	}
 
-	_, _, err = n.client.PostMessage(p1.ID,
-		slack.MsgOptionText(headerText, false),
-		slack.MsgOptionUsername("Disruption Status Bot"),
-		slack.MsgOptionIconURL("https://upload.wikimedia.org/wikipedia/commons/3/39/LogoChaosMonkeysNetflix.png"),
-		slack.MsgOptionBlocks(
-			headerBlock,
-			slack.NewDividerBlock(),
-			slack.NewSectionBlock(nil, disruptionBlocks, nil),
-			slack.NewDividerBlock(),
-			bodyBlock,
-		),
-		slack.MsgOptionAsUser(true),
-	)
-	if err != nil {
+	if nil != disruptionCron.Spec.Reporting && disruptionCron.Spec.Reporting.SlackChannel != "" {
+		if err := n.sendMessageToChannel(disruptionCron.Spec.Reporting.SlackChannel, slackMsg); err != nil {
+			logger.Warnw("slack notifier: couldn't send a message to the channel from the reporting", "slackChannel", disruptionCron.Spec.Reporting.SlackChannel, "error", err)
+		}
+	}
+
+	if err := n.sendMessageToUserChannel(disruptionCron, slackMsg, logger); err != nil {
+		logger.Warnw("slack notifier: couldn't send a message to the user", "error", err)
```

Comment:
We don't need this logged warning. The error will eventually be returned all the way to client-go's `tools/record/event.go#recordEvent` and will be logged as an error